guides/06_other-tutorials/theming-guide

```diff
@@ -0,0 +1,270 @@
+
+Similarly, to reference a core size, use the `*spacing_`, `*radius_`, or `*text_` prefix, followed by the size value. For example:
+
+```python
+theme = gr.themes.Default(radius_size="md").set(
+    button_primary_border_radius="*radius_xl",
+)
+```
+
+In the example above, we've set the `button_primary_border_radius` variable to `*radius_xl`. This variable will be set to the `xl` setting of the medium radius size range.
+
+#### Referencing Other Variables
+
+Variables can also reference each other. For example, look at the example below:
+
+```python
+theme = gr.themes.Default().set(
+    button_primary_background="#FF0000",
+    button_primary_background_hover="#FF0000",
+    button_primary_border="#FF0000",
+)
+```
+
+Having to set these values to a common color is a bit tedious. Instead, we can reference the `button_primary_background` variable in the `button_primary_background_hover` and `button_primary_border` variables, using a `*` prefix. 
+
+```python
+theme = gr.themes.Default().set(
+    button_primary_background="#FF0000",
+    button_primary_background_hover="*button_primary_background",
+    button_primary_border="*button_primary_background",
+)
+```
+
+Now, if we change the `button_primary_background` variable, the `button_primary_background_hover` and `button_primary_border` variables will automatically update as well.
+
+This is particularly useful if you intend to share your theme - it makes it easy to modify the theme without having to change every variable.
+
+Note that dark mode variables automatically reference each other. For example:
+
+```python
+theme = gr.themes.Default().set(
+    button_primary_background="#FF0000",
+    button_primary_background_dark="#AAAAAA",
+    button_primary_border="*button_primary_background",
+    button_primary_border_dark="*button_primary_background",
+)
+```
+
+`button_primary_border_dark` will draw its value from `button_primary_background_dark`, because dark mode always draw from the dark version of the variable.
+
+## Creating a Full Theme
+
+Let's say you want to create a theme from scratch! We'll go through it step by step - you can also see the source of prebuilt themes in the gradio source repo for reference - [here's the source](https://github.com/gradio-app/gradio/blob/main/gradio/themes/monochrome.py) for the Monochrome theme.
+
+Our new theme class will inherit from `gradio.themes.Base`, a theme that sets a lot of convenient defaults. We'll start by override the constructor. Let's we'll pass use the same constructor arguments, but change the defaults. Let's go for a green and blue colored theme. We'll keep the sizes of everything medium, and use the "Roboto" font from Google Fonts.
+
+```python
+from __future__ import annotations
+from typing import Iterable
+from gradio.themes.base import Base
+from gradio.themes.utils import colors, fonts, sizes
+
+
+class EmeraldBlue(Base):
```

Comment:
Might also be worth mentioning that you don't necessarily have to create a new class to create a new theme.

The following is less code and maybe easier for people to understand?

```python
emerald_blue = Base(
        primary_hue=colors.green,
        secondary_hue= colors.blue,
        neutral_hue=colors.gray,
        spacing_size=sizes.spacing_md,
        radius_size=sizes.radius_md,
        text_size=sizes.text_md)
```